syntax = "proto3";

package bluetooth.hal;

import "google/protobuf/empty.proto";

service HciHalFacade {
  rpc SendCommand(Command) returns (google.protobuf.Empty) {}
  rpc SendAcl(AclPacket) returns (google.protobuf.Empty) {}
  rpc SendSco(ScoPacket) returns (google.protobuf.Empty) {}
  rpc SendIso(IsoPacket) returns (google.protobuf.Empty) {}

  rpc StreamEvents(google.protobuf.Empty) returns (stream Event) {}
  rpc StreamAcl(google.protobuf.Empty) returns (stream AclPacket) {}
  rpc StreamSco(google.protobuf.Empty) returns (stream ScoPacket) {}
  rpc StreamIso(google.protobuf.Empty) returns (stream IsoPacket) {}
}

message Event {
  bytes payload = 1;
}

message Command {
  bytes payload = 1;
}

message AclPacket {
  bytes payload = 1;
}

message ScoPacket {
  bytes payload = 1;
}

message IsoPacket {
  bytes payload = 1;
}
